examples/tensorflow-akka/src/main/scala/modelserving/wine/WineModel.scala

```diff
@@ -0,0 +1,79 @@
+package modelserving.wine
+
+import java.nio.file.Path
+
+import scala.util.control.NonFatal
+
+import org.tensorflow.Tensor
+
+import modelserving._
+import modelserving.wine.avro._
+
+/**
+ * Loads [[WineModel]]s from a TensorFlow SavedModelBundle.
+ * @param savedModelBundlePath the path to the SavedModelBundle
+ * @param modelName the name of the model that will be used in ModelResultMetadata.
+ */
+object WineModelBundle {
+  def load(savedModelBundlePath: Path, modelName: String) = TensorFlowModelBundle.load(savedModelBundlePath, modelName, WineModel.apply)
+}
+
+/**
+ * Serves wine quality scores.
+ */
+final case class WineModel(val loadedModel: LoadedModel) extends TensorFlowModel[WineRecord, WineServingResult] {
+  val session = loadedModel.session
+  val signatures = loadedModel.signatures
+
+  def emptyServingResult = WineServingResult(.0)
+
+  /**
+   * Scores WineRecords.
+   */
+  def scoreWine(record: WineRecord): WineResult = {
+    val (servingResult, modelResultMetadata) = score(record)
+    WineResult(record, servingResult, modelResultMetadata)
+  }
+
+  /**
+   * invokes the TensorFlow bundled model for wine scoring.
+   */
+  def invokeModel(record: WineRecord): Either[String, WineServingResult] = {
+    try {
+      // Create record tensor
+      val modelInput = toTensor(record)
```

Comment:
It would be nice to store some result metadata like scoring time in millisecs.

Reply:
It does do that, look at `TensorFlowModel` `score` function. `The WineModel` extends that.